qt/python/mantidqt/mantidqt/widgets/sliceviewer/models/model.py

```diff
@@ -576,13 +576,16 @@ def _cut_names(self, cut: str):
 
         return xcut_name, ycut_name, help_msg
 
-    def get_hkl_from_xyz(self, xdim, ydim, zdim, xdata, ydata, zdata):
+    def get_hkl_from_full_point(self, full_point: List[float], qdims_i: List[int]):
+        """Gets the values of h, k and l from a full point which can include 3 or more dimensions."""
         basis_transform = self.get_proj_matrix()
-        if basis_transform is not None:
-            hkl = basis_transform[:, xdim] * xdata + basis_transform[:, ydim] * ydata + basis_transform[:, zdim] * zdata
-        else:
-            hkl = {0.0, 0.0, 0.0}
-        return hkl
+        if basis_transform is None:
+            return 0.0, 0.0, 0.0
+
+        # Get the values for the h, k and l projections
+        hkl_projection = tuple(full_point[q_i] for q_i in qdims_i)
```

Comment:
Really minor but I think `hkl_projection` is not a good name, perhaps  `q_xyz`? Recall this is the Q vector in the frame of the sliceviewer axes.